src/3rdParty/libE57Format/src/CheckedFile.cpp

```diff
@@ -44,6 +44,10 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
+#elif defined(__OpenBSD__)
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
 #elif defined( __APPLE__ )

```

Comment:
Sorry for the late comment, but this is also necessary for the other BSD's, so it might be worth to include them here as well. The [FreeBSD port has this patched already](https://github.com/freebsd/freebsd-ports/blob/main/cad/freecad/files/patch-src_3rdParty_libE57Format_src_CheckedFile.cpp) and it will probably be necessary for a future NetBSD port of FreeCAD.

Reply:
Ok, and what about DragonFly? Does it also need a special treatment or ist it handled by `__FreeBSD__`?

Reply:
They have their own [macro](https://www.dragonflybsd.org/docs/howtos/howtoporttodragonfly/#index7h3) (`__DragonFly__`), but since their ports are based on FreeBSD's they can compile with [FreeBSD as the build target](https://www.dragonflybsd.org/docs/howtos/howtoporttodragonfly/#index6h3).

Reply:
https://github.com/FreeCAD/FreeCAD/pull/16477

Reply:
Thank you @wwmayer :)